nodeadm/internal/api/net.go

```diff
@@ -39,7 +40,7 @@ func GetIMDSInstanceDetails(ctx context.Context, imdsClient *imds.Client) (*Inst
 }
 
 // Derive the default ClusterIP of the kube-dns service from EKS built-in CoreDNS addon
-func (details *ClusterDetails) GetClusterDns() (string, error) {
+func GetClusterDns(details *v1alpha1.ClusterDetails) (string, error) {
```

Comment:
What's the rationale for this? Seems like it should be working with the internal API types

Reply:
i wanted to work with the external types as much as possible to avoid having too an excess of duplicated types, but wasn't sure if it was a good idea to move this to the api package or just swap the signature like so